tests/rptest/services/provider_clients/azure_client.py

```diff
@@ -0,0 +1,99 @@
+from rptest.services.provider_clients.client_utils import query_instance_meta
+
+headers = {"Metadata-Flavor": "Microsoft"}
+
+
+class AzureClient:
+    """
+    Azure Client class.
+    Should implement function similar to AWS EC2 client
+    """
+    def __init__(self,
+                 region,
+                 key,
+                 secret,
+                 tenant,
+                 logger,
+                 endpoint=None,
+                 disable_ssl=True):
+
+        self._region = region
+        self._access_key = key
+        self._secret_key = secret
+        self._tenant = tenant
+        self._endpoint = endpoint
+        self._disable_ssl = disable_ssl
+        self._cli = self._make_client()
+
+    def _make_client(self):
+        # TODO: Work on Azure client
+        return None
```

Comment:
often better to `raise NotImplementedError` or something like that in these cases to fail faster if it's ever called. Or just delete the method since you'll get a nice error then too (including in the type checker).

Reply:
If I understand the flow correctly, this code _is_ actually called, and the code calling it expects to see a NoneType here to decide that this is Azure. Both sides of this could be improved.

Reply:
Yes, None here is a part of the flow. 'NotImplementedError' is better, though. We can improve that later